Library/Formula/diction.rb

```diff
@@ -1,14 +1,68 @@
-  sha1 '30c7c778959120d30fa67be9261d41de894f498b'
+  homepage "https://www.gnu.org/software/diction/"
+  url "http://ftpmirror.gnu.org/diction/diction-1.11.tar.gz"
+  mirror "https://ftp.gnu.org/gnu/diction/diction-1.11.tar.gz"
+  sha1 "30c7c778959120d30fa67be9261d41de894f498b"
 
   def install
     system "./configure", "--disable-debug", "--disable-dependency-tracking",
                           "--prefix=#{prefix}"
-    system "make install"
+    system "make", "install"
+  end
+
+  test do
+    ENV["LC_ALL"] = "C"
+    # Text from "The Elements of Style" Introductory by William Strunk, Jr.
+    # http://www.bartleby.com/141/strunk1.html
+    file = "strunk_intro.txt"
+    (testpath/file).write <<-EOS.undent
+    It is an old observation that the best writers sometimes disregard the
+    rules of rhetoric. When they do so, however, the reader will usually find in
+    the sentence some compensating merit, attained at the cost of the
+    violation. Unless he is certain of doing as well, he will probably do best
+    to follow the rules. After he has learned, by their guidance, to write
+    plain English adequate for everyday uses, let him look, for the secrets of
+    style, to the study of the masters of literature.
+    EOS
+
+    style_result =<<-EOS.undent
+    readability grades:
+            Kincaid: 9.9
+            ARI: 10.5
+            Coleman-Liau: 9.2
+            Flesch Index: 62.0/100 (plain English)
+            Fog Index: 13.3
+            Lix: 45.9 = school year 8
+            SMOG-Grading: 11.7
+    sentence info:
+            386 characters
+            86 words, average length 4.49 characters = 1.45 syllables
+            4 sentences, average length 21.5 words
+            50% (2) short sentences (at most 17 words)
+            0% (0) long sentences (at least 32 words)
+            1 paragraphs, average length 4.0 sentences
+            0% (0) questions
+            50% (2) passive sentences
+            longest sent 31 wds at sent 4; shortest sent 15 wds at sent 1
+    word usage:
+            verb types:
+            to be (2) auxiliary (2) 
+            types as % of total:
+            conjunctions 3(3) pronouns 9(8) prepositions 14(12)
+            nominalizations 5(4)
+    sentence beginnings:
+            pronoun (1) interrogative pronoun (1) article (0)
+            subordinating conjunction (2) conjunction (0) preposition (0)
+    EOS
+    assert_equal style_result, shell_output("#{bin}/style #{file}")
+
+    diction_result =<<-EOS.undent
+#{file}:2: When [they] do [so], [however], the reader will usually find in the sentence some compensating merit, attained at the cost of the violation.
+
+#{file}:5: After he has learned, by their guidance, to write plain English [adequate] for everyday uses, let him look, for the secrets of style, to the study of the masters of literature.
+
+4 phrases in 4 sentences found.
+    EOS
+    assert_equal diction_result, shell_output("#{bin}/diction #{file}")
```

Comment:
I don't think we need to match the contents exactly for two comments; it'll be a bit error prone. I think checking exit codes and/or partial output will be sufficient.